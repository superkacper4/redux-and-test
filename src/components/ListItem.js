import React from 'react'
import styled from 'styled-components'

const StyledListItem = styled.li`
    background-color: transparent;
    border-bottom: solid 2px white;
    color: white;
    width: 100%;
    height:10vh;
    margin:0;
    list-style-type: none;
    display: flex;
    justify-content: center;
    align-items:center;
    flex-direction: column;
    font-size: 2rem;
    position:relative;
`;

const StyledAdd = styled.button`
    height: 100%;
    width:10%;
    background-color: ${({ cart }) => cart ? 'red' : 'green'};
    border:none;
    position: absolute;
    top:0;
    right:0;
    border: solid 2px black;
    transition: background-color .5s;
    cursor: pointer;

    &:hover{
        background-color: ${({ cart }) => cart ? 'orange' : 'lightgreen'}
    }
`;

const ListItem = ({ children, cart, onClick }) => {
    return (
        <StyledListItem>
            <StyledAdd cart={cart} onClick={onClick}>k</StyledAdd>
            {children}
        </StyledListItem>
    )
}


export default ListItem;