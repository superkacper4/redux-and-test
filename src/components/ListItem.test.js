import React from 'react';
import { render } from '@testing-library/react'
import ListItem from './ListItem'

describe("AlertMessage", () => {
    let props = {
        children: "hello",
        cart: true,
        onClick: jest.fn().mockImplementation(() => false),
    };

    it("should match snapshot", () => {
        const { container } = render(<ListItem {...props} />);
        expect(container).toMatchSnapshot();
    });

    it("should close when close button is clicked", () => {
        const { getByText } = render(<ListItem {...props} />);

        fireEvent.click(getByText("k"));
        expect(props.onClick).toHaveBeenCalled();
        expect(props.onClick).toHaveBeenCalledWith(false);
    });
});


// describe('List item', () => {
//     it('Renders children', () => {
//         const { getByText } = render(
//             <ListItem>Hello</ListItem>
//         );
//         getByText("Hello");
//         expect(getByText).toMatchSnapshot();
//     })
// })
