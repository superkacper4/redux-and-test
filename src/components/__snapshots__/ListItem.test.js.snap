// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertMessage should match snapshot 1`] = `
<div>
  <li
    class="sc-AxjAm dSRkXJ"
  >
    <button
      class="sc-AxirZ gYOjaL"
    >
      k
    </button>
    hello
  </li>
</div>
`;

exports[`List item Renders children 1`] = `[Function]`;
